utilities/ovn-trace.c

```diff
@@ -154,11 +156,7 @@ main(int argc, char *argv[])
         }
 
         if (ovsdb_idl_has_ever_connected(ovnsb_idl)) {
-            if (!already_read) {
-                already_read = true;
-                read_db();
-            }
-
+            read_db();
```

Comment:
I didn't review this thoroughly but checking what read_db() does I see we'll be doing this at every iteration:
```
static void
read_db(void)
{
    read_datapaths();
    read_ports();
    read_mcgroups();
    read_address_sets();
    read_port_groups();
    read_gen_opts();
    read_flows();
    read_mac_bindings();
    read_fdbs();
}
```
That doesn't seem OK because we never cleanup any of the maps we populate there.  E.g., `read_datapaths()` will reinsert the same datapaths over and over again in the global `struct hmap datapaths` variable.

Reply:
I observe that the memory keeps growing if using the above code. I will re-optimize the code according to your advice.